src/database/schemamanager.cpp

```diff
@@ -43,24 +43,10 @@ bool SchemaManager::isBackwardsCompatibleWithVersion(int targetVersion) const {
     bool ok = false;
     int iBackwardsCompatibleVersion = backwardsCompatibleVersion.toInt(&ok);
 
-    // If the current backwards compatible schema version is not stored in the
-    // settings table, assume the current schema version is only backwards
-    // compatible with itself.
     if (backwardsCompatibleVersion.isNull() || !ok) {
-        // rryan 11/2010 We just added the backwards compatible flags, and some
```

Comment:
Why did your removed this part??

Reply:
It's an exception for an ancient version we definitely don't support anymore.

Reply:
@Holzhaus Why has this been merged? It is not possible to revoke a database version once it has been released. All exceptions in the upgrade path must be preserved forever.

Reply:
This is why I asked and expressed my concerns about this removal.

Reply:
@xerus2000 Unfortunately you not only documented the SchemaManager as advertised but slipped in functional changes. That's the main reason why I don't trust your "cleanup" PRs that might contain some hidden changes as this one.

Reply:
This would only affect people that upgrade directly from schema version 7 and want to downgrade to schema version 3 which isn't possible with the current Mixxx anyway, right? So you'd need to grab an old version of Mixxx, and in that case the code is still there. Or am I missing something?

Reply:
Anyway, I'll revert that change.

Reply:
I am also not sure. Whoever is removing this code should is in charge to explain in detail why this removal cannot affect any users of Mixxx. I'm not planning to remove this code, so I don't need to think about it. As simple as that.

Reply:
I reverted it: https://github.com/mixxxdj/mixxx/commit/a0e1228b96799b463c408c6d5c1fbc6d14457d06

Reply:
This only affected developers of mixxx who were using mixxx in a narrow timespan where schema version 7 was released without a min_compatible flag. Every version of mixxx you can get now includes the `min_compatible` flag in version 7, so this is superfluous. It is just confusing when reading the code and carries no use for anyone anymore. It is dead code.